wallet/udb/addressmanager.go

```diff
@@ -1303,6 +1303,62 @@ func (m *Manager) ImportPrivateKey(ns walletdb.ReadWriteBucket, wif *dcrutil.WIF
 	return managedAddr, nil
 }
 
+// ImportPubKey imports a compressed 33-byte serialized secp256k1 public key and
+// the derived P2PKH address.  This method may only be used by watching-only
+// wallets.
+func (m *Manager) ImportPublicKey(ns walletdb.ReadWriteBucket, pubkey []byte) (ManagedPubKeyAddress, error) {
+	defer m.mtx.Unlock()
+	m.mtx.Lock()
+
+	if !m.watchingOnly {
+		return nil, errors.E(errors.Invalid, "public keys may "+
+			"only be imported by watching-only wallets")
+	}
+
+	if len(pubkey) != secp256k1.PubKeyBytesLenCompressed {
+		return nil, errors.E(errors.Encoding, "invalid length for "+
+			"compressed pubkey")
+	}
+	switch pubkey[0] {
+	case secp256k1.PubKeyFormatCompressedEven,
+		secp256k1.PubKeyFormatCompressedOdd:
+	default:
+		return nil, errors.E(errors.Encoding, "invalid format byte "+
+			"for compressed pubkey")
+	}
+
+	// Prevent duplicates.
+	pkh := dcrutil.Hash160(pubkey)
+	if existsAddress(ns, pkh) {
+		return nil, errors.E(errors.Exist, "address for public key "+
+			"already exists")
+	}
+
+	// Encrypt public key.
+	encryptedPubKey, err := m.cryptoKeyPub.Encrypt(pubkey)
+	if err != nil {
+		return nil, errors.E(errors.Crypto,
+			errors.Errorf("encrypt imported pubkey: %v", err))
+	}
+
+	// Save the new imported address to the db and update start block (if
+	// needed) in a single transaction.
+	err = putImportedAddress(ns, pkh, ImportedAddrAccount,
```

Comment:
We have the following `AccountKind`s:
https://github.com/decred/dcrwallet/blob/d9945563749febf960601cc37210fb5051c63b09/wallet/addresses.go#L28-L41

Although `AccountKindImported` does not explicitly say *private* keys, that was what it was before this PR.

In `(*Wallet).KnownAddress`, there's the mapping:  https://github.com/decred/dcrwallet/blob/d9945563749febf960601cc37210fb5051c63b09/wallet/addresses.go#L244-L253

So as a user of that wallet method, how do we know if wallet has the private keys for that address or just the public keys.  Previously, we (dcrdex) could conclude that we have the private keys if the `KnownAddress` was `AccountKindBIP0044` or `AccountKindImported`.

Can we have a new `AccountKind` to indicate that it's imported without private keys?

Reply:
Is `MaxAccountNum + 1` the only account number that can be reserved?

Reply:
we are intentionally limiting this feature to only watching-only wallets to avoid this situation where we don't know if the private key is available or not

Reply:
But before this PR, doesn't `AccountKindImported` means an address added with import**priv**key, so *private*.
This feature uses the same account kind for import**pub**key

Reply:
Oh I'm sorry I didn't understand that this was a blocked RPC unless the wallet were watching only!